### ISSUE TITLE
Explore option of `globus` command starting an interactive REPL which uses ~/.globus_history

### ISSUE THREAD

**Issue by sirosen** [globus/globus-cli#98]
This is a bit of a wishlist feature.

One of the main `click` maintainers has a package which can do a basic REPL on arbitrary commands called [click-repl](https://github.com/click-contrib/click-repl).
However, this doesn't offer one of the important and attractive things about the REPL, which is the possibility of an isolated history. It also presents itself as a separate subcommand, whereas we might actually want it to be the default for our top-level command.
We do have some in-house experience writing our own captive shells, but that doubles as a cautionary tale -- we know how hard it is to get right.

One of the interesting notes about that implementation is that it maintains the same context object, which lets it act kind of like a "session". That means we could things like cache a user's bookmarks to avoid extra lookups, or store previously seen identity mappings and even tab complete them. It also exposes a rich featureset revolving around internal aliasing and variables.

In short, an interactive shell can be as complex as we find interesting and worthwhile.
However, this should not be an invitation to view this feature as necessary or by any means simple -- the number of questions it raises are significant.

**Comment by corpulentcoffee** [comment]
Also to capture it somewhere: if we were to actually add such a feature, one possibility might be to be able to enter a "context" in the style of Cisco IOS CLI as @jaswilli suggested.

```
Router(config)# interface serial 2
Router(config-if)# encapsulation frame-relay
Router(config-if)# interface serial 2.1
Router(config-subif)#
```
